### ISSUE TITLE
Manually Trigger Element Events

### ISSUE THREAD

**Issue by kdf2883** [ironmansoftware/universal-dashboard#616]
It would be nice to be able to trigger any of the element event's manually. For an example, I want to collapse an collapsible item after a user action. Kind of like this:

```
New-UDCollapsible -Items {
            New-UDCollapsibleItem -Id Collapse -Content{
                
                #Options
                New-UDCheckbox -Id "ChkBoxNone" -Label "Check Box"
                
                #Apply Button
                New-UDElement -Tag "a" -Content {"Apply"} -id btnApply -Attributes @{
                    onClick = {
                       #Collapse UDCollapsible
--->                   Get-UDElement -Id Collapse | Trigger-UDEvent -event OnClick  
                    }
                }
            }
        }

```

**Comment by BoSen29** [comment]
@adamdriscoll this should be solvable with Invoke-UDJavascript and a powershell wrapper. 
Do you want me to implement this, or guide the users to use Invoke-UDJavascript? 

I imagine something like: Trigger-UDEvent -id "Collapse" -event "OnClick" (which would just do the RestRequest for the "CollapseOnClick" endpoint) Should work?

**Comment by adamdriscoll** [comment]
Hmmm interesting. I wonder if invoking the REST endpoint directly might be hairy because you'll have to deal with HTTP headers and auth and stuff. You could just invoke the event manually with JS.  Maybe extending (via parameter set) or wrapping Invoke-UDJavaScript in another command (Trigger isn't an approved verb) would work. I'm a bit partial to having an overloaded parameter set to Invoke-UDJavaScript that accepts an ID and event name to trigger. Then we could invoke something like this:  https://stackoverflow.com/a/906565

**Comment by BoSen29** [comment]
Fair point, did not actually validate the approved verb list O:-) 
From my perspective, the Invoke-UDJavaScript is intended to allow for custom javascript execution and possibly an internal function for executing of JS without increasing the size of the clientside-package. 
Hence, i'd like to implement Invoke-UDEvent (WIP name) with the required "id" and "event" parameters (probably a validateset on the event with the implemented events) which in turn utilizes the Invoke-UDJavaScript in order to keep the function purely PowerShell for simplicity (with the easy solution you linked in stackoverflow ofc) 

I feel like the Invoke-UDEvent would be easier to grasp for beginners, and the name itself is self explanatory.

I can get a PR up on either solutions.

**Comment by adamdriscoll** [comment]
Ok. That works for me. Let's go with Invoke-UDEvent!

**Comment by adamdriscoll** [comment]
Implemented in 2.8
